network/netvisor/pn_vrouterbgp.py

```diff
@@ -0,0 +1,495 @@
+    else:
+        NEIGHBOR_EXISTS = False
+
+
+def run_cli(module, cli):
+    """
+    This method executes the cli command on the target node(s) and returns the
+    output. The module then exits based on the output.
+    :param cli: the complete cli string to be executed on the target node(s).
+    :param module: The Ansible module to fetch command
+    """
+    cliswitch = module.params['pn_cliswitch']
+    state = module.params['state']
+    command = get_command_from_state(state)
+
+    cmd = shlex.split(cli)
+
+    # 'out' contains the output
+    # 'err' contains the error messages
+    result, out, err = module.run_command(cmd)
+
+    print_cli = cli.split(cliswitch)[1]
+
+    # Response in JSON format
+    if result != 0:
+        module.exit_json(
+            command=print_cli,
+            stderr=err.strip(),
+            msg="%s operation failed" % command,
+            changed=False
+        )
+
+    if out:
+        module.exit_json(
+            command=print_cli,
+            stdout=out.strip(),
+            msg="%s operation completed" % command,
+            changed=True
+        )
+
+    else:
+        module.exit_json(
+            command=print_cli,
+            msg="%s operation completed" % command,
+            changed=True
+        )
+
+
+def get_command_from_state(state):
+    """
+    This method gets appropriate command name for the state specified. It
+    returns the command name for the specified state.
+    :param state: The state for which the respective command name is required.
+    """
+    command = None
+    if state == 'present':
+        command = 'vrouter-bgp-add'
+    if state == 'absent':
+        command = 'vrouter-bgp-remove'
+    if state == 'update':
+        command = 'vrouter-bgp-modify'
+    return command
+
+
+def main():
+    """ This portion is for arguments parsing """
+    module = AnsibleModule(
+        argument_spec=dict(
+            pn_cliusername=dict(required=False, type='str'),
+            pn_clipassword=dict(required=False, type='str', no_log=True),
+            pn_cliswitch=dict(required=False, type='str', default='local'),
+            state=dict(required=True, type='str',
+                       choices=['present', 'absent', 'update']),
+            pn_vrouter_name=dict(required=True, type='str'),
+            pn_neighbor=dict(type='str'),
+            pn_remote_as=dict(type='str'),
+            pn_next_hop_self=dict(type='bool'),
+            pn_password=dict(type='str', no_log=True),
+            pn_ebgp=dict(type='int'),
+            pn_prefix_listin=dict(type='str'),
+            pn_prefix_listout=dict(type='str'),
+            pn_route_reflector=dict(type='bool'),
+            pn_override_capability=dict(type='bool'),
+            pn_soft_reconfig=dict(type='bool'),
+            pn_max_prefix=dict(type='int'),
+            pn_max_prefix_warn=dict(type='bool'),
+            pn_bfd=dict(type='bool'),
+            pn_multiprotocol=dict(type='bool',
```

Comment:
This isn't a `bool` if it accepts choices=['ipv4-unicast', 'ipv6-unicast']

I think you mean `type:'str'`